### PR TITLE
Update python nodes SIGINT handling

### PR DESCRIPTION
**Pull request by ivanpauno**
Related to https://github.com/ros2/rclpy/pull/844.

### CODE REVIEW

File: test_communication/test/action_client_py.py [hunk 1]
```diff
@@ -190,7 +192,10 @@ def is_result_valid(get_result_response):
     except BaseException:
         print('Exception in action client:', file=sys.stderr)
         raise
+    except ExternalShutdownException:
+        print('Action client stopped with sigterm', file=sys.stderr)
+        rc = 1
```
Comment (hidmic) [inline test_communication/test/action_client_py.py:197]: @ivanpauno `BaseException` will always match `ExternalShutdownException`, `except` clauses need reordering.

**Review by hidmic — APPROVED** [review]
LGTM pending green CI

File: test_communication/test/action_client_py.py (hunk 1 above)
Comment (ivanpauno) [reply test_communication/test/action_client_py.py:194]: ah I didn't pay attention to that, thanks!
Fixed in https://github.com/ros2/system_tests/pull/490/commits/238321a89daf6a1351629c9bc9bd4bbff58f896f.

File: test_communication/test/action_server_py.py [hunk 2]
```diff
@@ -180,6 +182,8 @@ def execute_goal(goal_handle):
     except BaseException:
         print('Exception in action server:', file=sys.stderr)
         raise
+    except ExternalShutdownException:
+        sys.exit(1)
```
Comment (hidmic) [inline test_communication/test/action_server_py.py:186]: @ivanpauno this one has the same ordering issue!

Comment (ivanpauno) [reply test_communication/test/action_server_py.py]: argh, fixed in https://github.com/ros2/system_tests/pull/490/commits/9f0bb2a7ff36c54d67c50b53627f5edc9456a5a3.

**Review by hidmic — APPROVED** [review]

**Pull request merged**